s2spy/_base_calendar.py

```diff
@@ -303,10 +363,60 @@ def visualize_interactive(
         if utils.bokeh_available():
             # pylint: disable=import-outside-toplevel
             from ._bokeh_plots import bokeh_visualization
+
             return bokeh_visualization(self, n_years, relative_dates)
         return None
 
     @property
     def flat(self) -> pd.DataFrame:
         """Returns the flattened intervals."""
         return self.get_intervals().stack()  # type: ignore
+
+
+class Period(ABC):
+    """Basic construction element of calendar for defining target period."""
+
+    def __init__(self, length: str, gap: str = "0d", target: bool = False) -> None:
+        self.length = DateOffset(**self._parse_time(length))
+        self.gap = DateOffset(**self._parse_time(gap))
+        self.target = target
```

Comment:
I think these should be properties, or have setters/getters implemented. The following example is accepted, but will break `calendar.append`:
```py
precursor_1 = s2spy.time.PrecursorPeriod("10d", "10d")
precursor_1.length = "14d"
```

Reply:
Good idea! I will make them "read-only" attributes with property.